extensions/azure/cosmos/contract-definition-store-cosmos/src/main/java/org/eclipse/dataspaceconnector/contract/definition/store/CosmosContractDefinitionStore.java

```diff
@@ -61,40 +62,17 @@ public CosmosContractDefinitionStore(CosmosDbApi cosmosDbApi, TypeManager typeMa
         this.retryPolicy = retryPolicy;
         this.partitionKey = partitionKey;
         lockManager = new LockManager(new ReentrantReadWriteLock(true));
+        queryResolver = new ReflectionBasedQueryResolver<>(ContractDefinition.class);
```

Comment:
this should get passed in through the CTor, so that the `ReflectionBasedQueryResolver` can be moved to core

Reply:
I think it's related to your previous comment: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/930#discussion_r833382897
Why we should move it to core-base? Are modules allowed to depend on core-base?

Reply:
what I meant was to _pass this as constructor argument_ rather than _instantiating directly_. That way, the `CosmosContractDefinitionStore` only has a dependency on `QueryResolve`.
Then, the `ReflectionBasedQueryResolver` instance could be `@Inject`-ed in the `CosmosContractDefinitionStoreExtension`. This would of course require some extension to have registered it previously. 

The more I think about this, the more I think the way it is now is actually better, because is much less hassle and it hides the implementation. So please feel free to disregard :)